src/Connection.php

```diff
@@ -262,7 +311,10 @@ public function fetchOne(
     ) : array|false
     {
         $sth = $this->perform($statement, $values);
-        return $sth->fetch(PDO::FETCH_ASSOC);
+        /** @var array<array-key, mixed> $result */
```

Comment:
Might this be good as a custom type, e.g. `fetch_assoc_array mixed[]` and the the @return type would be `fetch_assoc_array|false` ?